packages/wrapped-keys/src/lib/litActions/solana/src/signTransactionWithSolanaEncryptedKey.js

```diff
@@ -0,0 +1,60 @@
+const {
+  clusterApiUrl,
+  Connection,
+  Keypair,
+  Transaction,
+} = require('@solana/web3.js');
+
+(async () => {
+  const LIT_PREFIX = 'lit_';
+
+  let decryptedPrivateKey;
+  try {
+    decryptedPrivateKey = await Lit.Actions.decryptToSingleNode({
+      accessControlConditions,
+      chain: 'ethereum',
+      ciphertext,
+      dataToEncryptHash,
+      authSig: null,
+    });
+  } catch (error) {
+    Lit.Actions.setResponse({
+      response: `Error decrypting data to private key: ${error.message}`,
+    });
+    return;
+  }
+
+  if (!decryptedPrivateKey) {
+    // Exit the nodes which don't have the decryptedData
+    return;
+  }
+
+  const privateKey = decryptedPrivateKey.startsWith(LIT_PREFIX)
+    ? decryptedPrivateKey.slice(LIT_PREFIX.length)
+    : decryptedPrivateKey;
+  const solanaKeyPair = Keypair.fromSecretKey(
+    Uint8Array.from(Buffer.from(privateKey, 'hex'))
+  );
+
+  try {
+    const transaction = Transaction.from(
+      Buffer.from(serializedTransaction, 'base64')
```

Comment:
Same as in the above file, why buffer to begin with?

Reply:
The Solana library pretty much only intakes `Uint8Array`s. `serializedTransaction` is being passed to the Lit Action as a hex string and is being converted to a Buffer here as required

Reply:
Would it make sense to update the `signTransactionWithEncryptedKey` param to take Uint8Arrays?

Reply:
If the serialized Solana tx are uint8Arrays we should standardize to accept Uint8Array instead of string even for Eth tx

Reply:
We're passing it as a Json value so can't be a Uint8Array we need to convert it to either a string or native array